datasets/preprocess_utils.py

```diff
@@ -243,3 +243,181 @@ def bin_stream_hilmi24(
+                 4.94284725e-03,  1.80499806e-03]
+
+    elif version=='edr3':
+        # EDR3: Notice that the last two parameters (describing the
+        # r-offset curvature due to background issues) are much smaller.
+        pars = [ 2.61815727e+00,  2.69372875e+00,  1.45644592e+00, -5.99023051e-04,
+                 3.97535324e-01,  3.15794343e-01,  2.55484718e-01,  1.00000000e-05,
+                 8.30152817e-04, -3.57980758e-04]
+
+        fudgeShift = 2.4e-3 # Additive shift in peak (mag)
+    else:
+        raise Exception("Unrecognized Gaia version: %s"%version)
+
+    i10g,i10r,i10i,i10z = pars[0:4]
+    kg,kr,ki,kz = pars[4:8]
+    r1,r2 = pars[8:10]
+    desFlux = (kg * fluxg * (1.0 + fnuPrime[:, 0] * i10g) +
+               kr * fluxr * (1.0 + fnuPrime[:, 1] * i10r) +
+               ki * fluxi * (1.0 + fnuPrime[:, 2] * i10i) +
+               kz * fluxz * (1.0 + fnuPrime[:, 3] * i10z))
+    mGDES = -2.5 * np.log10(desFlux)
+    rMag = -2.5 * np.log10(fluxr)
+    mGDES += r1 * (rMag - 17.0) + r2 * (rMag - 17.0)**2.
+    mGDES += fudgeShift # Peak shift
+    return mGDES
+
+def get_decam_g_r_i_z(iso_name = 'Dotter', iso_age = 11.5, iso_metallicity = 0.00016, iso_distance_modulus = 16.807, iso_stellar_mass = 2e4):
+    """Get simulated DECam g, r, i, z magnitudes from the AAU isochrone"""
+    
+    # Simulated AAU isochrone for g and r bands 
+    iso_g_r = isochrone.factory(name=iso_name,
+                            age=iso_age,  # Gyr
+                            metallicity=iso_metallicity,  # Approximate Z value for the stream metallicity
+                            distance_modulus=iso_distance_modulus,  # Average distance modulus from the stream properties
+                            )
+
+    # Simulated AAU isochrone for i and z bands
+    iso_i_z = isochrone.factory(name=iso_name,
+                            age=iso_age,  # Gyr
+                            metallicity=iso_metallicity,  # Approximate Z value for the stream metallicity
+                            distance_modulus=iso_distance_modulus,  # Average distance modulus from the stream properties
+                            band_1 = 'i',
+                            band_2 = 'z'
+                            )
+
+    decam_g, decam_r = iso_g_r.simulate(stellar_mass=iso_stellar_mass)
+    decam_i, decam_z = iso_i_z.simulate(stellar_mass=iso_stellar_mass)
+
+    return decam_g, decam_r, decam_i, decam_z
+
+def get_gaia_g(decam_g, decam_r, decam_i, decam_z):
+    """Get simulated Gaia G band magnitudes from the AAU isochrone"""
+
+    gaia_g = gaia_transform(decam_g, decam_r, decam_i, decam_z, version='edr3')
+
+    filtered_gaia_g = []
+    
+    color = decam_g - decam_r + 0.04
+    min_color = 0.2
+    max_color = max(color)
+
+    min_mag = 14.875
+    max_mag = 19.512
+
+    for c, m, j in zip(color, decam_r, gaia_g):
+            if min_color <= c <= max_color and min_mag <= m <= max_mag:
+                    filtered_gaia_g.append(j)
```

Comment:
This block has inconsistent indentation compared the the rest of the code. Please use 4-space identication like the rest of the code.

Reply:
fixed